front/src/applications/operationalStudies/components/MacroEditor/osrdToNge.ts

```diff
@@ -293,9 +317,20 @@ const importTimetable = async (
   // Create one NGE train run per OSRD train schedule
   let labelId = 0;
   const trainruns: Trainrun[] = trainSchedules.map((trainSchedule) => {
-    let formatedLabels: Label[] = [];
+    let formatedLabels: (Label | undefined)[] = [];
+    let trainrunFrequency: TrainrunFrequency | undefined;
     if (trainSchedule.labels) {
       formatedLabels = trainSchedule.labels.map((label) => {
+        if (label.includes('frequency')) {
```

Comment:
The code which handle labels looks fine overall. However, I think that using `forEach` loop might be more appropriate than map in this case. Since not all labels need to be transformed into a new Object, especially the `frequency` labels, `forEach` would allow processing each label without returning a value, avoiding the insertion of `undefined` into the list of labels.

That said, it work as-is, so I'm fine with leaving it as is if preferred.

Reply:
You're right, I'll change it.

Reply:
Done